File: Data Structures & Algorithms/merge-intervals/submission-3.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()

        ans = [intervals[0]]
        for i in range(1, len(intervals)):
            s1, e1 = ans[-1]
            s2, e2 = intervals[i]
            if s2 <= e1:
                ans[-1] = [s1, max(e1, e2)]
            else:
                ans.append([s2, e2])
        return ans
        

File: Data Structures & Algorithms/trapping-rain-water/submission-3.java
class Solution {
    public int trap(int[] height) {
        int ans = 0;

        int l = 0; int r = height.length-1;
        int leftMax = height[l];
        int rightMax = height[r];

        while (l < r) {
            if (leftMax < rightMax) {
                l++;
                leftMax = Math.max(leftMax, height[l]);
                ans += (leftMax - height[l]);
            } else {
                r--;
                rightMax = Math.max(rightMax, height[r]);
                ans += (rightMax - height[r]);
            }
        }

        return ans;
    }
}
